src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -95,21 +96,41 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
     }
 
     public async deleteTreeItemImpl(): Promise<void> {
-        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
-        await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
+        if (this.parent.server) {
+            const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+            await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
+        } else {
+            const username_connString = nonNullProp(this.parent.connectionString, 'username');
+            const password_connString = nonNullProp(this.parent.connectionString, 'password');
+            const host = nonNullProp(this.parent.connectionString, 'hostName');
+            const config = { user: username_connString, password: password_connString, host, port: 5432 };
+            dropdb(config, this.databaseName, (err) => {
+                if (err) {
+                    console.error(err);
+                    process.exit(-1);
```

Comment:
Is there a reason you did this instead of throwing an error like we normally do?

Reply:
Thought callback option was the only option available with this function. I checked the library, actually callback is not necessary as promise API is available as well.